### PR TITLE
FAQ overhaul

### PR DESCRIPTION
**Pull request by bgrant0607**
Take into account the larger scope of kpt and eliminate some of the smaller technical details.

cc @justinsb @selfmanagingresource

### CODE REVIEW

File: site/faq/README.md [hunk 1]
```diff
@@ -4,59 +4,45 @@
 
 Please visit the [roadmap document] and the [kpt milestones].
 
-### What does kpt provide that git clone doesn't?
-
-kpt enables out-of-the-box workflows that `git clone` does not. Such as: cloning
-and versioning git subdirectories, updating from upstream by performing
-structured merges on resources, programmatically editing configuration (rather
-than with an editor), etc.
```
Comment (yuwenma) [inline site/faq/README.md:12]: Is this removal intended?
This section refers to the `kpt pkg` which is not described anywhere below. And git is so popular so it seems useful to tell the differences (even if it's verbose). Or do we want to de-emphasize the package management feature from kpt?

Comment (bgrant0607) [reply site/faq/README.md:12]: Yes, this removal was intended. This was the main emphasis when we first created the kpt packaging mechanism, but isn't now. For instance, we plan to support container images as storage.

I'm trying to up-level the FAQ to the whole toolchain and the wysiwyg experience.

File: site/faq/README.md [hunk 2]
```diff
@@ -4,59 +4,45 @@
...
-- Allows you to edit the configuration in-place without creating complex patches.
-- Focuses on rebase with resource merge strategy allowing for edited config to
+- Optimizes for WYSIWYG configuration and in-place customization.
+- Allows edits to the configuration in-place without creating complex patches.
+- Supports rebase with resource merge strategy allowing for edited config to
   be updated.
-- Has a continuous learning curve as you usually start small with modifying
-  several YAML files using an editor and then want to scale with complexity of
-  the application.
 - Enables workflows that combine programmatic changes (functions) with manual
```
Comment (yuwenma) [inline site/faq/README.md:32]: nit: square brackets around `functions`

Comment (bgrant0607) [reply site/faq/README.md]: Thanks. Looks like that was a preexisting bug. Should be fixed now.

File: site/faq/README.md (hunk 1 above)
Comment (yuwenma) [reply site/faq/README.md:12]: ah, that makes sense.

File: site/faq/README.md [hunk 3]
```diff
@@ -4,59 +4,45 @@
...
-(pkg, fn, live) independently if needed. For example:
+transformation, and actuation functionality. At the same time, kpt follows a modular
+design principle to make it possible to use each of its functionality
+independently if needed. For example:
 
 - You can use packaging without declaring functions
-- You can use imperative functions to operate on vanilla directories of k8s
+- You can use imperative functions to operate on vanilla directories of Kubernetes
   resources
 - You can use apply logic without buying into full the packaging story (still
```
Comment (yuwenma) [inline site/faq/README.md:47]: oops, "the full packaging story"

**Review by yuwenma — APPROVED** [review]

File: site/faq/README.md (hunk 3 above)
Comment (bgrant0607) [reply site/faq/README.md]: Thanks. Fixed
